woocommerce/woocommerce-gutenberg-products-block
# Fix pattern route performance

PR linked with https://github.com/woocommerce/woocommerce/pull/41168


## What

Fixes #

## Why



This PR splits the `/patterns` endpoint into different endpoints to allow the client to parallelize the requests. The new endpoints are:

`/private/ai/images` -> generate images and return the client the image URLs.
`/private/ai/products` -> generate the payload to pass to /ai/product.
`/private/ai/product` -> update a specific product.
`/private/ai/patterns` -> generate and save patterns.


In the next iteration, we should improve how we handle the errors, but for now, these endpoints are private, so it should not be a big issue. 

## Testing Instructions




_Please consider any edge cases this change may have, and also other areas of the product this may impact._

- Install and activate [this version](https://github.com/woocommerce/woocommerce-blocks/files/13248734/woocommerce.zip) of WooCommerce.
- Install and activate WooCommerce Blocks from the following zip file: [woocommerce-gutenberg-products-block.zip](https://github.com/woocommerce/woocommerce-blocks/files/13248736/woocommerce-gutenberg-products-block.zip)
- Install and activate Jetpack.
- In your wp-admin, you should see the following section for Jetpack:

<img width="1232" alt="Screenshot 2023-10-22 at 09 29 08" src="https://user-images.githubusercontent.com/15730971/267093030-fc804759-f724-4fa2-829d-7bfd441962ec.png">

- Click on "Set up Jetpack" and make sure the connection is successful:

<img width="1232" alt="Screenshot 2023-10-22 at 09 29 08" src="https://user-images.githubusercontent.com/15730971/267093371-661ee4c5-a0ac-40fd-9e5e-835a0e4c1cb7.png">

- Install and activate WooCommerce Beta Tester (the plugin is available [within the monorepo](https://github.com/woocommerce/woocommerce/tree/trunk/plugins))
- Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag:

<img width="1233" alt="Screenshot 2023-10-22 at 10 32 24" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/816867c6-34e5-40a1-a87d-f4a7cda46429">

### Test the AI-managed images

1. Head over to Home > Customize your Store:
 
<img width="1232" alt="Screenshot 2023-10-22 at 09 29 08" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/b5d34131-3edc-47ed-a8e4-8e2775ca1bbd">

2. Click on design with AI.
3. Type a description for your business.
4. Make sure the AI content is properly generated and images are assigned to patterns and products based on the description provided.


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Add suggested changelog entry here.
